app/cdap/components/WranglerGrid/SelectColumnPanel/DataTable/index.tsx

```diff
@@ -0,0 +1,182 @@
+import T from 'i18n-react';
+import TableRowWidget from 'components/WranglerGrid/SelectColumnPanel/DataTable/TableRow';
+import { IColumnTableProps } from 'components/WranglerGrid/SelectColumnPanel/DataTable/types';
+import {
+  ADD_TRANSFORMATION_PREFIX,
+  SELECT_COLUMN_LIST_PREFIX,
+} from 'components/WranglerGrid/SelectColumnPanel/constants';
+import { MULTI_SELECTION_COLUMN } from 'components/WranglerGrid/SelectColumnPanel/constants';
+import { TableContainer, Table, TableHead, TableRow, TableCell } from '@material-ui/core';
+import { grey } from '@material-ui/core/colors';
+import styled from 'styled-components';
+import { NormalFont, SubHeadBoldFont } from 'components/common/TypographyText';
+import { NoDataSVG } from 'components/GridTable/iconStore';
+
+const SelectColumnTableContainer = styled(TableContainer)`
+  height: 100%;
+`;
+
+const SelectColumnTable = styled(Table)`
+  display: flex;
+  flex-direction: column;
+`;
+
+const SelectColumnTableHead = styled(TableHead)`
+  height: 54px;
+`;
+
+const SelectColumnTableRow = styled(TableRow)`
+  font-weight: 400;
+  font-size: 16px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: ${grey[700]};
+  display: grid;
+  grid-template-columns: 8% 45% 45%;
+  align-items: center;
+  height: 100%;
+`;
+
+const SelectColumnTableHeadCell = styled(TableCell)`
+  &.MuiTableCell-head {
+    padding: 0;
+    font-weight: 400;
+    font-size: 16px;
+    line-height: 150%;
+    letter-spacing: 0.15px;
+    color: ${grey[900]};
+    border-bottom: none !important;
+  }
+`;
+
+const FlexWrapper = styled(Box)`
+  display: flex;
+  height: 100%;
+  align-items: center;
+`;
+
+const CenterAlignBox = styled(Box)`
+  text-align: center;
+`;
+
+export default function({
+  columns,
+  transformationDataType,
+  onSingleSelection,
+  selectedColumns,
+  dataQualityValue,
+  isSingleSelection,
+  handleDisableCheckbox,
+  onMultipleSelection,
+  setSelectedColumns,
+  transformationName,
+}: IColumnTableProps) {
+  const handleChange = () => {
+    if (
+      MULTI_SELECTION_COLUMN?.filter(
+        (option) => option.value === transformationName && option.isMoreThanTwo === false
+      ).length > 0
+    ) {
+      if (selectedColumns.length) {
+        setSelectedColumns([]);
+      } else {
+        if (columns?.length > 2) {
+          setSelectedColumns(columns.slice(0, 2));
+        } else {
+          setSelectedColumns(columns);
+        }
+      }
+    } else {
+      if (selectedColumns?.length) {
+        setSelectedColumns([]);
+      } else {
+        setSelectedColumns(columns);
+      }
+    }
+  };
+
+  const columnsToDisplay = transformationDataType?.includes('all')
+    ? columns
+    : columns.filter((eachColumn) =>
+        transformationDataType?.includes(eachColumn?.type[0]?.toLowerCase())
+      );
+
+  return (
+    <SelectColumnTableContainer data-testid="column-table-parent">
+      {columnsToDisplay.length === 0 ? (
+        <FlexWrapper>
+          <CenterAlignBox>
```

Comment:
no large blocks of code in ternary operators

Reply:
Done